servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java

```diff
@@ -161,8 +163,13 @@ private static void populateRoutes(final ExecutionContext executionContext,
             final String path = entry.getKey();
             final ServiceAdapterHolder adapterHolder = entry.getValue().buildRoute(executionContext);
             final StreamingHttpService route = closeable.append(adapterHolder.adaptor());
-            verifyNoOverrides(allRoutes.put(path, adapterHolder.serviceInvocationStrategy()
-                    .offloadService(executionContext.executor(), route)), path, emptyMap());
+            verifyNoOverrides(allRoutes.put(path,
+                            StreamingHttpServiceToOffloadedStreamingHttpService.offloadService(
+                                    adapterHolder.serviceInvocationStrategy(),
+                                    executionContext.executor(),
+                                    () -> Thread.currentThread() instanceof IoThreadFactory.IoThread,
```

Comment:
should this take into account `isIoThreadSupported` (same question for all other usages)? Seems like we should have a utility method somewhere (e.g. `io.servicetalk.transport.api`?) to share code as this repeats in a few areas.

```java
   static BooleanSupplier shouldOffload(IoExecutor executor) {
        return executor.isIoThreadSupported() ?
                () -> Thread.currentThread() instanceof IoThreadFactory.IoThread : Boolean.TRUE::booleanValue;
    }
```

Reply:
Yes, I would be willing. There was such a method in an earlier revision but it was banished. 😊It seems common enough to include.On key goal is to be able to use a method reference rather than a lambda so that the lambda is self-identifying in the debugger.